backend/internal/server/serverstream/serverstream.go

```diff
@@ -0,0 +1,29 @@
+package serverstream
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+)
+
+// ServerStreamWithContext is a grpc.ServerStream that allows its context to be modified.
+// It does so by overriding the Context() function of grpc.ServerStream.
```

Comment:
nit, but should this be overwrite?

```suggestion
// It does so by overwriting the Context() function of grpc.ServerStream.
```

Reply:
Ah yea, I think most people use these terms loosely, but I think you're right given that `override` seems to be an OOP concept.